src/faction_camp.cpp

```diff
@@ -546,6 +547,22 @@ recipe_id base_camps::select_camp_option( const std::map<recipe_id, translation>
         return pos_name_ids.front();
     }
 
+    std::string swap_name;
+    recipe_id swap_id;
+
+    for( size_t i = 0; i < pos_names.size() - 1; i++ ) {
+        for( size_t k = i + 1; k < pos_names.size(); k++ ) {
+            if( localized_compare( pos_names[k], pos_names[i] ) ) {
+                swap_name = pos_names[k];
+                swap_id = pos_name_ids[k];
+                pos_names[k] = pos_names[i];
+                pos_name_ids[k] = pos_name_ids[i];
+                pos_names[i] = swap_name;
+                pos_name_ids[i] = swap_id;
```

Comment:
```suggestion
    for( size_t i = 0; i < pos_names.size() - 1; i++ ) {
        for( size_t k = i + 1; k < pos_names.size(); k++ ) {
            if( localized_compare( pos_names[k], pos_names[i] ) ) {
                std::swap( pos_name_ids[i], pos_name_ids[k] );
                std::swap( pos_name[i], pos_name[k] );
```
It is especially important when swapping strings or other containers since they have very efficient zero-copy implementations. Currently it copies strings three times for each swap.
I would recommend to do the same in https://github.com/CleverRaven/Cataclysm-DDA/pull/45647 but it's not so critical for `size_t`.

Reply:
Thanks. That's good practice, and so should be used in both cases. I don't think it's critical in this case either, as we're operating on human speed here (the time is stopped and the player interacts with the UI), but there is no reasonable reason not to apply good practices when it doesn't lead to a fair bit less readable code (and in this case it improves readability).